Basic template defined

kamakay opened the pull request:
### Contributing guide
https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/README.md

### Changelog
* Added base template for Spark on Suse
### Description of the change

azurecla commented:
Hi __@kamakay__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <span>You've already signed the contribution license agreement. Thanks!</span>
        <span>We will now validate the agreement and then real humans will evaluate your PR.</span>

TTYL, AZPRBOT;

kamakay commented:
@EgoAleSum : test are green, can I merge on my local repository my current branch to master and send it on next push?
Thx

kamakay commented:
For reviewwers: can i merge locally our branches to local master or should we take features (on branches) separated ?

EgoAleSum commented:
@kamakay  Ciao Roberto, I don't have the ability to merge your PR, sorry.

@bmoore-msft Can you please check? Thanks!

kamakay commented:
Dear All,
What can I do now? I followed the github guide for PR

bmoore-msft commented on spark-on-suse/README.md:
```diff
@@ -0,0 +1,28 @@
+</a>
+
+
+How to run this template from Powershell
```
update these instructions to use the deployment scripts in the root of the repo: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/README.md#deploying-samples

kamakay replied:
Ok, just added basic setup

bmoore-msft commented on spark-on-suse/azuredeploy.json:
```diff
@@ -0,0 +1,257 @@
+      "type": "securestring",
+	  "metadata": { "description": "The LInux VM password" }
+    },
+    "sparkMasterSize": {
```
make this a variable - size suitable/recommended for your sample

kamakay replied:
Done

bmoore-msft commented on spark-on-suse/azuredeploy.json:
```diff
@@ -0,0 +1,257 @@
+  },
+  "variables": {
+    "resourceGroupLocation": "[resourceGroup().location]",
+	"vmStorageAccountContainerName": "vhd",
```
spacing

kamakay replied:
Done

bmoore-msft commented on spark-on-suse/azuredeploy.json:
```diff
@@ -0,0 +1,257 @@
+	"vmStorageAccountContainerName": "vhd",
+    "storageAccountName": "[concat(uniquestring(resourceGroup().id),'storageacc')]",
+    "virtualMachines_spark01_name": "sparkMaster",
+	"sparkMasterPublicIpName": "sparkMasterPublicIp",
```
spacing

kamakay replied:
Done

bmoore-msft commented on spark-on-suse/azuredeploy.json:
```diff
@@ -0,0 +1,257 @@
+    "sparkMasterNetworkInterfaceName": "ni_master",
+    "networkInterfaces_slave1_name": "ni_slave1",
+    "sparkMasterInternalIP": "10.0.0.4",
+    "scriptUrl": "https://raw.githubusercontent.com/Azure/azure-quickstart-templates/master/spark-on-suse/",
```
See: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md#samples-that-contain-extra-artifacts-custom-scripts-nested-templates-etc for BP on including artifacts.

kamakay replied:
Done using artifact variables

bmoore-msft commented on spark-on-suse/azuredeploy.json:
move this to the top of the resource definition

bmoore-msft commented on spark-on-suse/azuredeploy.json:
move up

bmoore-msft commented on spark-on-suse/azuredeploy.json:
remove

bmoore-msft commented on spark-on-suse/azuredeploy.json:
remove

bmoore-msft commented on spark-on-suse/azuredeploy.json:
remove

bmoore-msft commented on spark-on-suse/azuredeploy.json:
remove

bmoore-msft commented on spark-on-suse/azuredeploy.json:
remove the empty properties

bmoore-msft commented on spark-on-suse/azuredeploy.json:
remove

bmoore-msft commented on spark-on-suse/azuredeploy.json:
remove

bmoore-msft commented on spark-on-suse/azuredeploy.json:
remove

bmoore-msft commented on spark-on-suse/azuredeploy.json:
remove

bmoore-msft commented on spark-on-suse/azuredeploy.json:
remove

bmoore-msft commented on spark-on-suse/azuredeploy.json:
update apiVersion

bmoore-msft commented on spark-on-suse/azuredeploy.json:
See: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md#samples-that-contain-extra-artifacts-custom-scripts-nested-templates-etc for handling artifacts

kamakay commented on spark-on-suse/azuredeploy.json:
Done

kamakay commented on spark-on-suse/azuredeploy.json:
Removed

kamakay commented on spark-on-suse/azuredeploy.json:
Done

kamakay commented on spark-on-suse/azuredeploy.json:
Removed

kamakay commented on spark-on-suse/azuredeploy.json:
Removed

kamakay commented on spark-on-suse/azuredeploy.json:
Removed

kamakay commented on spark-on-suse/azuredeploy.json:
Removed

kamakay commented on spark-on-suse/azuredeploy.json:
Done

kamakay commented on spark-on-suse/azuredeploy.json:
Done

kamakay commented on spark-on-suse/azuredeploy.json:
Done

kamakay commented on spark-on-suse/azuredeploy.json:
Done

kamakay commented on spark-on-suse/azuredeploy.json:
Done

kamakay commented on spark-on-suse/azuredeploy.json:
Done

kamakay commented on spark-on-suse/azuredeploy.json:
Done

kamakay commented:
@bmoore-msft I have added "_artifactsLocationSasToken" parmater with blank value on the parameters json.
I receive an error from Travis but I added a blank value
"AssertionError: spark-on-suse/azuredeploy.parameters.json -  Parameter "_artifactsLocationSasToken" is missing its value field"

kamakay commented:
Hi, from my point of view everything is correct.

kamakay commented:
build seems locked

kamakay commented:
@bmoore-msft : Can you please check? Thanks :)

bmoore-msft commented on spark-on-suse/README.md:
````diff
@@ -0,0 +1,38 @@
+Simple execute the script and pass in the folder name of the sample you want to deploy.  For example:
+
+```PowerShell
+.\Deploy-AzureResourceGroup.ps1 -ResourceGroupLocation 'eastus' -ArtifactsStagingDirectory '[foldername]'
````
Since this sample has artifacts that need to be staged, just provide the cmd line example that sets the switch (i.e. the second one)

kamakay replied:
Hi, can you make it more clear? could this be correct
**.\Deploy-AzureResourceGroup.ps1 -ResourceGroupLocation 'eastus' -ArtifactsStagingDirectory 'spark-on-suse' -UploadArtifacts**

bmoore-msft replied:
you have it exactly...  so remove the generic stuff and just put the cmd you have here...

bmoore-msft commented on spark-on-suse/azuredeploy.json:
```diff
@@ -0,0 +1,335 @@
+    },
+    "adminPassword": { "type": "securestring",
+      "metadata": {
+        "description": "The LInux VM password"
```
typo (capital I in linux)

kamakay replied:
Done :)

bmoore-msft commented on spark-on-suse/azuredeploy.json:
make this a child resource of the VM it's running on...

kamakay commented on spark-on-suse/azuredeploy.json:
ok for this one, there is another i would implement but I need to not include it into xxx VM resources cause it depends on another custom script, external to the xxx VM

kamakay commented:
@bmoore-msft: from my point of view everything is ok, can I merge please? Thanks :)

kamakay commented on spark-on-suse/azuredeploy.json:
Done

kamakay commented:
@bmoore-msft: Hi, can you please check? From our point of view everything is ok :)

bmoore-msft commented on spark-2.0-on-suse/azuredeploy.json:
```diff
@@ -0,0 +1,416 @@
+        "description": "The Linux VM password"
+      }
+    },
+	"numberOfSlaves" : {
```
just remove this altogether and use the default value for the variable you have below...

kamakay replied:
Hi, we would like to allow users to specify these parameters (usr, passsword and slaves numeber)

bmoore-msft commented on spark-2.0-on-suse/azuredeploy.json:
move dependsOn above properties

bmoore-msft commented on spark-2.0-on-suse/azuredeploy.json:
move dependsOn above properties

bmoore-msft commented on spark-2.0-on-suse/azuredeploy.json:
indentation is off here

kamakay commented on spark-2.0-on-suse/azuredeploy.json:
Done

kamakay commented on spark-2.0-on-suse/azuredeploy.json:
Done

kamakay commented on spark-2.0-on-suse/azuredeploy.json:
Done

bmoore-msft commented on spark-2.0-on-suse/README.md:
```diff
@@ -0,0 +1,38 @@
+</a>
+
+
+## Deploying Samples
```
remove generic instructions for using the script and just provide the specific example needed starting at line 32

kamakay replied:
Ok, took the example on quickstart root

bmoore-msft commented on spark-2.0-on-suse/azuredeploy.json:
```diff
@@ -0,0 +1,419 @@
+      "metadata": {
+      "description": "Number Of Slaves"
+      },
+      "defaultValue": 1
```
if we really need this for the sample is there a min & max we can add here then?

kamakay replied:
Sure :)

kamakay replied:
Ok

bmoore-msft commented on spark-2.0-on-suse/azuredeploy.json:
```diff
@@ -0,0 +1,419 @@
+    "scriptUrl": "[parameters('_artifactsLocation')]",
+    "ScriptFolder" : "scripts",
+    "sparkInstallScriptName": "install_spark_environment.sh",
+    "numberOfSlaves": "[parameters('numberOfSlaves')]"
```
remove this if we're keeping the parameter

kamakay replied:
Done

bmoore-msft replied:
It's still showing for me @ line 59

kamakay replied:
Ok now (refactoring issue)

kamakay commented:
@bmoore-msft : can you merge? From my pount of view we are ready ☺
